### PR TITLE
Update code examples from "classic" operators to taskflow

### PR DESCRIPTION
**Pull request by hankehly**
Related: #25319

### Summary

This PR updates the following documentation about classic operators (`PythonOperator`, `PythonVirtualenvOperator`, etc..) to their TaskFlow decorator alternatives.

* How-to Guides / Using Operators / PythonOperator
* Upgrading from 1.10 to 2

### How to check the updates

1. Build the docs
```
breeze build-docs --docs-only --package-filter apache-airflow
```
2. Open `docs/_build/docs/apache-airflow/latest/index.html` in your browser

### CODE REVIEW

File: docs/apache-airflow/howto/operator/python.rst [hunk 1]
```diff
@@ -22,8 +22,11 @@
 PythonOperator
```
Comment (hankehly) [inline docs/apache-airflow/howto/operator/python.rst:22]: We could go a step further and retitle this page "@task"

**Comment by hankehly** [comment]
(waiting for checks to pass before requesting review)

File: docs/apache-airflow/howto/operator/python.rst [hunk 2]
```diff
@@ -22,8 +22,11 @@
...
 ==============
 
-Use the :class:`~airflow.operators.python.PythonOperator` to execute
-Python callables.
+Use the ``@task`` decorator to execute Python callables.
+
+.. warning::
+    Previous versions of Airflow required using :class:`~airflow.operators.python.PythonOperator`
+    to execute Python callables. The ``@task`` decorater is now the recommended method.
```
Comment (uranusjr) [inline docs/apache-airflow/howto/operator/python.rst]: The last version to _require_ using PythonOperator was 1.10.10; I’d probably rewrite this to

> The `@task` decorator is recommended over the more “classic” `PythonOperator` approach to execute Python callables.

Comment (hankehly) [reply docs/apache-airflow/howto/operator/python.rst]: Thanks. Please see [`fc66a4e`](https://github.com/apache/airflow/pull/25845/commits/fc66a4efb5d418e0aaee583406aa1fe9b3421621)

**Review by uranusjr — APPROVED** [review]

**Comment by uranusjr** [comment]
Need to fix static check errors.

**Comment by hankehly** [comment]
@uranusjr 
Static checks fixed in https://github.com/apache/airflow/pull/25845/commits/fad7585dcbfcd3eaf136a7db70f25b157e90cae6

**Review by kaxil — APPROVED** [review]

File: docs/apache-airflow/upgrading-from-1-10/index.rst [hunk 3]
```diff
@@ -924,6 +929,23 @@ instead. On top of that, a new dag.callback_exceptions counter metric has
...
+For details, see :doc:`/tutorial_taskflow_api`.
+
+============================= ============================================
+Classic Operator              TaskFlow Decorator
+============================= ============================================
+``PythonOperator``            ``@task`` (short for ``@task.python``)
+``PythonVirtualenvOperator``  ``@task.virtualenv``
+``BranchPythonOperator``      ``@task.branch``
+``DockerOperator``            ``@task.docker``
+============================= ============================================
```
Comment (kaxil) [inline docs/apache-airflow/upgrading-from-1-10/index.rst:947]: We also have ``@task.kubernetes`` now

Comment (kaxil) [reply docs/apache-airflow/upgrading-from-1-10/index.rst:947]: it was added in https://github.com/apache/airflow/pull/25663

Comment (hankehly) [reply docs/apache-airflow/upgrading-from-1-10/index.rst:947]: Thanks, let's add that in

Comment (hankehly) [reply docs/apache-airflow/upgrading-from-1-10/index.rst:947]: @kaxil Added in [dfd4327](https://github.com/apache/airflow/pull/25845/commits/dfd43274474c85bd76bf301254e1c1e2d7c7db49)

**Pull request merged**